### PR TITLE
🌱 cron: pass config as an argument to binaries (4/n)

### PR DESCRIPTION
**Pull request by spencerschrock**
#### What kind of change does this PR introduce?

Part of a series of refactors to generalize the cron infrastructure to enable re-use by the [ossf criticality score project](https://github.com/ossf/criticality_score)

- [X] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### What is the current behavior?
The `cron/internal/config/config.yaml` file is embedded inside cron binaries, and packaged with the various docker files.
This was not flexible for re-using the cron infrastructure across projects.

#### What is the new behavior (if this is a feature change)?**
1. All cron binaries which interact with the `config` package now support a `config` flag.
```
  --config string
        Location of config file (default "/etc/scorecard/config.yaml")
```

2. The config file is now volume mounted for k8s cronjobs at `/etc/scorecard/config.yaml`

- [X] Tests for the changes have been added (for bug fixes/features)

#### Which issue(s) this PR fixes
NONE

#### Special notes for your reviewer
I made a few naming choices, (e.g. storing configs at `/etc/scorecard`), feel free to nit and I can patch.

#### Does this PR introduce a user-facing change?

For user-facing changes, please add a concise, human-readable release note to
the `release-note`

(In particular, describe what changes users might need to make in their
application as a result of this pull request.)

```release-note
NONE
```

### CODE REVIEW

File: cron/internal/bq/main.go [hunk 1]
```diff
@@ -31,6 +32,8 @@ import (
 	"github.com/ossf/scorecard/v4/cron/internal/data"
 )
 
+var configFilename = flag.String(config.ConfigFlag, config.ConfigDefault, config.ConfigUsage)
+
```
Comment (azeemshaikh38) [inline cron/internal/bq/main.go:36]: Can define this flag in `config.go` so that multiple definitions are not required?

Comment (azeemshaikh38) [reply cron/internal/bq/main.go:36]: Basically,

```
// config.go

var configFileFlag = ...

func ReadConfig() err {
  // Read input file from `--config`
}
```

```
// main.go

flag.Parse()
if err := config.ReadConfig(); err != nil {
  // handle error
}
```

File: cron/internal/config/config.go [hunk 2]
```diff
@@ -35,6 +33,14 @@ const (
 	ShardNumFilename string = ".shard_num"
 	// TransferStatusFilename file identifies if shard transfer to BigQuery is completed.
 	TransferStatusFilename string = ".transfer_complete"
+
+	// ConfigFlag is the name of the CLI flag to specify a config file.
+	ConfigFlag string = "config"
+	// ConfigDefault is the default value for the config file if not provided as a CLI arg.
+	ConfigDefault string = "/etc/scorecard/config.yaml"
+	// ConfigUsage is a description of the config CLI flag.
+	ConfigUsage string = "Location of config file"
```
Comment (azeemshaikh38) [inline cron/internal/config/config.go:42]: Need not expose these consts if we are defining the flag in this file.

Also, let's keep the default value of `ConfigDefault` to empty "" and make the flag required to ensure is being passed in.

**Review by azeemshaikh38 — APPROVED** [review]

**Comment by oliverchang** [comment]
@calebbrown FYI

File: cron/internal/bq/main.go (hunk 1 above)
Comment (spencerschrock) [reply cron/internal/bq/main.go:36]: I think it makes the `flag.Parse()` call opaque, but it would work

**Review by azeemshaikh38 — APPROVED** [review]

**Review by azeemshaikh38 — APPROVED** [review]

File: cron/k8s/controller.yaml [hunk 3]
```diff
@@ -52,13 +52,21 @@ spec:
           containers:
           - name: controller
             image: gcr.io/openssf/scorecard-batch-controller:stable
-            args: ["cron/internal/data/projects.csv"]
+            args: ["--config=/etc/scorecard/config.yaml", "cron/internal/data/projects.csv"]
```
Comment (azeemshaikh38) [inline cron/k8s/controller.yaml:55]: Since the controller uses `os.Args[1]`, the argument positions should be swapped?

**Review by azeemshaikh38 — APPROVED** [review]

File: cron/k8s/controller.yaml (hunk 3 above)
Comment (spencerschrock) [reply cron/k8s/controller.yaml:55]: The controller uses `reader = localFiles(flag.Args())` where 
> flag.Args returns the non-flag command-line arguments.

**Pull request merged**